android/src/app/organicmaps/downloader/DownloaderAdapter.java

```diff
@@ -42,36 +40,46 @@
 import java.util.Stack;
 
 class DownloaderAdapter extends RecyclerView.Adapter<DownloaderAdapter.ViewHolderWrapper>
-                     implements StickyRecyclerHeadersAdapter<DownloaderAdapter.HeaderViewHolder>
 {
-  private static final int HEADER_ADVERTISMENT_ID = CountryItem.CATEGORY__LAST + 1;
-  private static final int HEADER_ADS_OFFSET = 10;
-
   private static final int TYPE_COUNTRY = 0;
+  private static final int TYPE_HEADER = 1;
 
   private static final String DOWNLOADER_MENU_ID = "DOWNLOADER_BOTTOM_SHEET";
 
   private final RecyclerView mRecycler;
   private final Activity mActivity;
   private final DownloaderFragment mFragment;
-  private final StickyRecyclerHeadersDecoration mHeadersDecoration;
-
   private boolean mMyMapsMode = true;
   private boolean mSearchResultsMode;
   private String mSearchQuery;
 
+  private final List<GenericItem> mItemsAndHeader = new ArrayList<>();
+
   private final List<CountryItem> mItems = new ArrayList<>();
   // Core Country.id String -> List<CountryItem> mapping index for updateItem function.
   // Use List, because we have multiple search results now for a single country.
   private final Map<String, List<CountryItem>> mCountryIndex = new HashMap<>();
 
-  private final SparseArray<String> mHeaders = new SparseArray<>();
   private final Stack<PathEntry> mPath = new Stack<>();  // Holds navigation history. The last element is the current level.
-  private int mNearMeCount;
 
   private int mListenerSlot;
   private CountryItem mSelectedItem;
 
+  private static class GenericItem {
+    @Nullable
+    public final String mHeaderText;
+    @Nullable
+    public final CountryItem mItem;
+    public GenericItem(@Nullable CountryItem item) {
+      mItem = item;
+      mHeaderText = null;
+    }
+    public GenericItem(@Nullable String headerText) {
```

Comment:
nit: braces formatting )

Reply:
Is there a way to make the PR lint job to detect those issues? Its easy to forget about this when you are used to a certain style

Reply:
We don't like autoformatters, and reformatting pollutes the history. that makes it a bit complicated.